tsc/meetings/2020-06-04.md

```diff
@@ -0,0 +1,72 @@
+Minutes from 51st OpenVDB TSC meeting, June 4th, 2020, (GMT)
+
+Attendees: *Nick* A., *Jeff* L., *Ken* M., *Dan* B.
+
+Additional Attendees: Johannes Meng (Intel), JT Nelson (Blender),
+Peter Cheng (DW), Andre Pradhana (DW), Bruce Chernia (Intel),
+Ibrahim Sani Kache (DW)
+
+Regrets: *Peter* C.
+
+Agenda:
+
+1) Confirm quorum
+2) Secretary
+3) Fast Sweeping
+4) Python 3
+5) Other
+6) Next meeting
+
+
+1) Quorum was confirmed.
+
+2) Secretary was Nick Avramoussis
+
+3) Fast Sweeping
+
+Ken has pushed up his implementation of his approach to the re-computation of
+signed distance fields, Fast Sweeping:
+http://www.museth.org/Ken/Publications_files/Museth_SIG17.pdf
+The code is available on Ken's fork of OpenVDB:
+https://github.com/kmuseth/openvdb/tree/OVDB-133_FastSweeping
+Ken to open a draft pull request into VDB for others to comment on and trial the
+implementation. Ken, some work still to be done however the bulk of the
+algorithm is complete. There exists a variety of new public methods to perform
+re-computations of level sets, conversions of scalar fields (signed fog) to
+level sets and the value extrapolation of level sets, potentially coupled with
+another scalar grid. CI build is currently failing due to some strict C++
+checks which should be easy to fix. It was noted that the extension of anything
+other than a scalar grid (i.e. vector grid) currently requires multiple
+invocations of the algorithm, something that could be improved.
+
+4) Python 3
+
+Ibrahim asked about python 3 support and OpenVDB. We currently have CI images
+that run in VFX 2020 which test with 2020's python version, currently at 3.7.
+There have been some reported issues with python 3.8, however this should only
+be specific to the architecture of the unit tests and not actually the python
+plugin itself. TSC unaware of any other issues regarding python 3. Note that the
+VFX 2021 container has now be drafted.
+https://vfxplatform.com/
+
+5) Other
+
+Discussion into OpenVDB's dependencies. Ken, one of the biggest complaints he
+hears is VDB's dependency list. Having some of these dependencies as VFX
+platform candidates could help - mainly referring to blosc. Jeff, LLVM (as a
+requirement of AX) should not really be a part of the VFX platform (due to its
+volatility w.r.t API/ABI), with downstream software instead opting for always
```

Comment:
One option we are considering is ASWF specific forks of problematic libraries, but we haven't taken any decisions or actions on that. LLVM was discussed, but our preliminary thoughts, in the context of OSL, were that we felt it was too volatile, as mentioned here, to take on attempting to track it.